### PR TITLE
Add TensorFlow sample

### PR DESCRIPTION
**Pull request by viliam-durina**

### CODE REVIEW

File: tensorflow-integration/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+After this, `data` subdirectory should be created with the following contents:
+
+```
+data/model/1/saved_model.pb
+data/model/1/variables/variables.data-00000-of-00001
+data/model/1/variables/variables.index
+data/imdb_word_index.json
+```
+
+Now, you can run either local or remote classification:
```
Comment (eminn) [inline tensorflow-integration/README.md]: Can we add a clarification about what is local and remote classification ?

File: tensorflow-integration/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+As a sample, we took the IMDB Movie Reviews Dataset as provided by the
+TensorFlow Keras Datasets. We train the model as described in a
+[tutorial](https://www.tensorflow.org/tutorials/keras/basic_text_classification)
+and save it.
+
+To run this sample, you need to have python installed. Since TF 1.12
+[doesn't support the current python
+version](https://github.com/tensorflow/tensorflow/issues/17022), you
+should use python 3.6 and install tensorflow using;
```
Comment (eminn) [inline tensorflow-integration/README.md]: Do we also need to list java 8 and maven as prerequisite ?

Comment (viliam-durina) [reply tensorflow-integration/README.md]: These are prerequisites of Jet, I don't think it needs to be mentioned here. Nor do we list a computer and a power socket :).

File: tensorflow-integration/README.md (hunk 1 above)
Comment (viliam-durina) [reply tensorflow-integration/README.md]: i'll fix this

File: pom.xml [hunk 3]
```diff
@@ -46,6 +46,7 @@
         <module>tf-idf</module>
         <module>wordcount</module>
         <module>jdbc</module>
+        <module>tensorflow-integration</module>
```
Comment (cangencer) [inline pom.xml]: I think just `tensorflow` is enough

File: tensorflow-integration/src/main/java/WordIndex.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.Serializable;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.Map;
+
+public class WordIndex implements Serializable {
```
Comment (cangencer) [inline tensorflow-integration/src/main/java/WordIndex.java]: missing javadoc

**Comment by cangencer** [comment]
Also missing mention in the readme in the root

**Comment by viliam-durina** [comment]
verify

File: tensorflow/README.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+Each way has pros and cons and discussion of them is beyond the scope of
+this sample.
+
+As a sample, we took the Large Movie Reviews Dataset as provided by the
+TensorFlow Keras Datasets. We trained a model as described in a
+[tutorial](https://www.tensorflow.org/tutorials/keras/basic_text_classification)
+and save it. The result of the training is in the
+[`data/model`](data/model) subdirectory.
+
```
Comment (eminn) [inline tensorflow/README.md]: Could we please put a few lines about what is the actual classification done here along with the input and output types ?

File: tensorflow/README.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+```
+$ mvn compile exec:java -Dexec.mainClass=InProcessClassification -Dexec.args="data"
+```
+
+To run the classification on a model server, you'll need Docker
+installed. It's possible to create a model server using the following
+command:
+```
+docker run -p 8500:8500 --mount  \
+type=bind,source={hazelcast-jet-code-samples}/tensorflow/data/model,target=/models/reviewSentiment \
```
Comment (eminn) [inline tensorflow/README.md]: Can we replace `{hazelcast-jet-code-samples}/tensorflow/data/model` with `$(pwd)/data/model` ?

File: tensorflow/src/main/java/InProcessClassification.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+            Pipeline p = Pipeline.create();
+            p.drawFrom(Sources.map(reviewsMap))
+             .map(Map.Entry::getValue)
+             .mapUsingContext(modelContext, (model, review) -> executeClassification(wordIndex, model, review))
+             // TensorFlow executes modes in parallel, we'll use 2 local threads to maximize throughput.
+             .setLocalParallelism(2)
+             .drainTo(Sinks.logger());
+
+            instance.newJob(p).join();
```
Comment (eminn) [inline tensorflow/src/main/java/InProcessClassification.java]: Could we improve logging to make the pipeline output distinguishable from the regular logs ?

**Pull request merged**